### PR TITLE
[Chore][Manifest] flip 7 elementwise ops to spec-only

### LINKED ISSUE #1153: [META][Manifest] flip 7 elementwise ops to spec-only (ctor drift from canonical)
### Symptom / Motivation

Audit (#1143) of the 31 ops with `status: implemented` against the canonical Op pattern in `docs/design/ops-design.md` found 7 elementwise ops whose ctor signatures fail to conform: shape-tuple parameters and no `kernel_map=`/`tune=` kwargs. Per `.claude/rules/manifest-trust-model.md` ("Implementation does not conform to spec → set `status: spec-only`, fix code in follow-up PR"), these manifest entries are mis-marked and must be reverted to `spec-only` before any code-level alignment lands.

### Root Cause Analysis

The 7 ops were aligned at earlier points in time, before the canonical elementwise pattern (`UnaryOp` / `BinaryOp` / `FusedGatedOp` in `tileops/ops/elementwise.py`) and the kwargs-only `kernel_map=` / `tune=` ctor convention were established. None inherits the canonical family bases; each implements its own multi-input-broadcast ctor that takes shape tuples. Concrete drift, identical across all 7:

- ctor takes shape tuples (e.g. `input: tuple, mask: tuple, value: tuple`) instead of scalar `N_total: int`;
- ctor missing `kernel_map=` and `tune=` kwargs;
- kernel built directly (e.g. `WhereFwdKernel(self.N_total, dtype)`) rather than via `dispatch_kernel(kernel_map)` + `kernel_map[_op_name](...)`.

The remaining 24 implemented ops (14 attention, RMSNormFwdOp, 7 reduction, 2 scan) all carry `kernel_map=` / `tune=` and conform — verified by mechanical scan over `inspect.signature(<Op>.__init__)` for every entry with `status: implemented`.

### Related Files

- `tileops/manifest/elementwise_unary_activation.yaml`
- `tileops/manifest/elementwise_binary.yaml`
- `tileops/manifest/elementwise_multi_input.yaml`
- `tileops/ops/elementwise.py` (reference only, NOT edited in this PR)
- `.claude/rules/manifest-trust-model.md` (governing rule)

## Goal

Flip `status: implemented → spec-only` for these 7 manifest entries — no other change in this PR:

- `ClampFwdOp`
- `ClampScalarFwdOp`
- `ClampMinFwdOp`
- `ClampMaxFwdOp`
- `MaskedFillFwdOp`
- `MaskedFillScalarFwdOp`
- `WhereFwdOp`

Manifest-only PR. Code alignment (adding `kernel_map=` / `tune=` kwargs, normalizing the ctor against `UnaryOp` / `BinaryOp` / `FusedGatedOp`, plus a family-level multi-input-broadcast template) belongs to a follow-up PR tracked in #1142 wave 3.

## Plan

1. Locate each of the 7 ops in the relevant `tileops/manifest/elementwise_*.yaml` shard (one of `_unary_activation.yaml`, `_binary.yaml`, `_multi_input.yaml`).
2. Edit only the `status` field from `implemented` to `spec-only` using a round-trip parser (`ruamel.yaml`) so comments and key order are preserved.
3. Run `python scripts/validate_manifest.py` and confirm zero new errors.
4. Confirm `git diff` touches only `tileops/manifest/elementwise_*.yaml` and only the `status` line for the 7 listed entries.
5. Run pytest collection (`python -m pytest --collect-only -q`) to confirm no test discovery breaks.

## Constraints

- **Manifest-only PR.** Per `.claude/rules/manifest-trust-model.md`: "A PR that implements/modifies ops or kernels must not also modify the manifest." This PR therefore must NOT touch `tileops/ops/`, `tileops/kernels/`, `tests/`, `benchmarks/`, or design docs.
- Edit only the `status` field on these 7 entries — do not modify `signature`, `roofline`, `source`, `workloads`, or any other field.
- Do not change YAML file structure, key order, or comments — round-trip parser only.
- Do not flip any other op's status (the remaining 24 implemented ops are conformant per the audit).

## Acceptance Criteria

- [ ] AC-1: Modified files pass unit tests.
- [ ] AC-2: All 7 listed ops show `status: spec-only` after the PR; verifiable by:
  ```
  python -c "from tileops.manifest import load_manifest; m=load_manifest(); print({n: m[n]['status'] for n in ['ClampFwdOp','ClampScalarFwdOp','ClampMinFwdOp','ClampMaxFwdOp','MaskedFillFwdOp','MaskedFillScalarFwdOp','WhereFwdOp']})"
  ```
  Every value must be `spec-only`.
- [ ] AC-3: `git diff main -- tileops/` is non-empty for `tileops/manifest/elementwise_*.yaml` and empty for every other path under `tileops/`.
- [ ] AC-4: `python scripts/validate_manifest.py` exits 0.
- [ ] AC-5: No other op's `status` is changed; `git diff main -- tileops/manifest/ | grep -E '^[+-][[:space:]]*status:' | sort -u` shows only `-  status: implemented` and `+  status: spec-only`, and only on the 7 entries.

### PR DESCRIPTION
**Pull request by lcy-seso**
Closes #1153

## Summary

- Flip `status: implemented → spec-only` for 7 elementwise ops across 3 manifest shards: `ClampFwdOp`, `ClampScalarFwdOp`, `ClampMinFwdOp`, `ClampMaxFwdOp` (`elementwise_unary_activation.yaml`); `MaskedFillFwdOp`, `MaskedFillScalarFwdOp` (`elementwise_binary.yaml`); `WhereFwdOp` (`elementwise_multi_input.yaml`).
- Manifest-only change. No edits under `tileops/ops/`, `tileops/kernels/`, `tests/`, `benchmarks/`, or `docs/`. Round-trip parser preserved comments and key order.

## Test plan

- [x] AC-1: Modified files pass unit tests. (`python -m pytest -q tests/test_validate_manifest.py tests/test_ops_manifest.py tests/test_workloads_to_params.py` → 217 passed)
- [x] AC-2: All 7 ops report `status: spec-only` via `load_manifest()` lookup.
- [x] AC-3: `git diff main -- tileops/` is non-empty only for `tileops/manifest/elementwise_*.yaml`; no other path under `tileops/` is touched.
- [x] AC-4: `python scripts/validate_manifest.py` exits 0 (`All manifest checks passed.`).
- [x] AC-5: `git diff main -- tileops/manifest/ | grep -E '^[+-][[:space:]]*status:' | sort -u` shows only `- status: implemented` and `+ status: spec-only`, applied to the 7 entries only.

## Additional context

Manifest-trust-model rule: an implementation does not yet conform to spec → flip `status` to `spec-only`. Code/tests will be re-aligned in follow-up PRs.

### CODE REVIEW

File: tileops/manifest/elementwise_binary.yaml [hunk 1]
```diff
@@ -60,7 +60,7 @@ MaskedFillFwdOp:
   # value form lands as MaskedFillScalarFwdOp below per the
   # "No Optional[Tensor]" manifest rule for variant splits.
   family: elementwise
-  status: implemented
+  status: spec-only
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_binary.yaml:63]: Out of scope for #1153 (status-flip-only); follow-up will document non-conformance per .claude/rules/manifest-trust-model.md.

Current HEAD: 4a0112a

File: tileops/manifest/elementwise_binary.yaml [hunk 2]
```diff
@@ -100,7 +100,7 @@ MaskedFillScalarFwdOp:
   # Per the "No Optional[Tensor]" manifest rule, this is split from the
   # 0-dim-Tensor-value primary entry above.
   family: elementwise
-  status: implemented
+  status: spec-only
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_binary.yaml:103]: Out of scope for #1153 (status-flip-only); follow-up will document non-conformance per .claude/rules/manifest-trust-model.md.

Current HEAD: 4a0112a

File: tileops/manifest/elementwise_multi_input.yaml [hunk 3]
```diff
@@ -13,7 +13,7 @@ WhereFwdOp:
   # constrains them to ``same_as(input)``; once the kernel grows an input
   # cast path, relax ``other`` to the full float union and document the
   # promotion contract here.
-  status: implemented
+  status: spec-only
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_multi_input.yaml:16]: Out of scope for #1153 (status-flip-only); follow-up will document non-conformance per .claude/rules/manifest-trust-model.md.

Current HEAD: 4a0112a

File: tileops/manifest/elementwise_unary_activation.yaml [hunk 4]
```diff
@@ -400,7 +400,7 @@ ClampFwdOp:
   # single-bound min-only form, and the single-bound max-only form each
   # land as their own variant_of entries below.
   family: elementwise
-  status: implemented
+  status: spec-only
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_unary_activation.yaml:403]: Out of scope for #1153 (status-flip-only); follow-up will document non-conformance per .claude/rules/manifest-trust-model.md.

Current HEAD: 4a0112a

File: tileops/manifest/elementwise_unary_activation.yaml [hunk 5]
```diff
@@ -438,7 +438,7 @@ ClampScalarFwdOp:
   # with Number (or None) bounds. Per the "No Optional[Tensor]" manifest
   # rule, this is split from the Tensor-bound primary entry above.
   family: elementwise
-  status: implemented
+  status: spec-only
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_unary_activation.yaml:441]: Out of scope for #1153 (status-flip-only); follow-up will document non-conformance per .claude/rules/manifest-trust-model.md.

Current HEAD: 4a0112a

File: tileops/manifest/elementwise_unary_activation.yaml [hunk 6]
```diff
@@ -477,7 +477,7 @@ ClampMinFwdOp:
   # Single-bound Tensor variant: torch.clamp_min(input, min: Tensor) —
   # equivalent to torch.clamp with only the lower bound provided.
   family: elementwise
-  status: implemented
+  status: spec-only
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_unary_activation.yaml:480]: Out of scope for #1153 (status-flip-only); follow-up will document non-conformance per .claude/rules/manifest-trust-model.md.

Current HEAD: 4a0112a

File: tileops/manifest/elementwise_unary_activation.yaml [hunk 7]
```diff
@@ -512,7 +512,7 @@ ClampMaxFwdOp:
   # Single-bound Tensor variant: torch.clamp_max(input, max: Tensor) —
   # equivalent to torch.clamp with only the upper bound provided.
   family: elementwise
-  status: implemented
+  status: spec-only
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_unary_activation.yaml:515]: Out of scope for #1153 (status-flip-only); follow-up will document non-conformance per .claude/rules/manifest-trust-model.md.

Current HEAD: 4a0112a

File: tileops/manifest/elementwise_binary.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_binary.yaml:63]: Out of scope for #1153 (status-flip-only); follow-up will document non-conformance per .claude/rules/manifest-trust-model.md.

Current HEAD: 4a0112a

File: tileops/manifest/elementwise_binary.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_binary.yaml:103]: Out of scope for #1153 (status-flip-only); follow-up will document non-conformance per .claude/rules/manifest-trust-model.md.

Current HEAD: 4a0112a

File: tileops/manifest/elementwise_unary_activation.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_unary_activation.yaml:403]: Out of scope for #1153 (status-flip-only); follow-up will document non-conformance per .claude/rules/manifest-trust-model.md.

Current HEAD: 4a0112a

File: tileops/manifest/elementwise_unary_activation.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_unary_activation.yaml:441]: Out of scope for #1153 (status-flip-only); follow-up will document non-conformance per .claude/rules/manifest-trust-model.md.

Current HEAD: 4a0112a

File: tileops/manifest/elementwise_multi_input.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_multi_input.yaml:16]: Out of scope for #1153 (status-flip-only); follow-up will document non-conformance per .claude/rules/manifest-trust-model.md.

Current HEAD: 4a0112a

File: tileops/manifest/elementwise_unary_activation.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_unary_activation.yaml:480]: Out of scope for #1153 (status-flip-only); follow-up will document non-conformance per .claude/rules/manifest-trust-model.md.

Current HEAD: 4a0112a

File: tileops/manifest/elementwise_unary_activation.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tileops/manifest/elementwise_unary_activation.yaml:515]: Out of scope for #1153 (status-flip-only); follow-up will document non-conformance per .claude/rules/manifest-trust-model.md.

Current HEAD: 4a0112a

**Review by Ibuki-wind — APPROVED** [review]
Clean — no issues.
